### PR TITLE
Use The_Comms_Arena in particle communication.

### PR DESCRIPTION
**Pull request by atmyers**
The proposed changes:
- [ ] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

**Comment by WeiqunZhang** [comment]
Because `The_Comms_Arena` is an alias to `The_Pinned_Arena` when gpu aware mpi is not used, the code like below might be problematic.
```
    // assuming ParallelDescriptor::UseGpuAwareMpi() is false                                                                                                                

    amrex::PODVector<char, PolymorphicArenaAllocator<char> > rcv_buffer;
    rcv_buffer.setArena(The_Comms_Arena()); // pinned memory if not using gpu aware mpi                                                                                      

    if (ParallelDescriptor::UseGpuAwareMpi())
    {
        // ...                                                                                                                                                               
    }
    else
    {
        ...
        // rcv_buffer is not device memory                                                                                                                                   
        Gpu::htod_memcpy_async(rcv_buffer.dataPtr(), pinned_rcv_buffer.dataPtr(), pinned_rcv_buffer.size());
    }
```

**Comment by atmyers** [comment]
Maybe we can use `The_Comms_Arena()` with gpu-aware MPI, and `The_Arena()` without it.

**Review by ax3l — APPROVED** [review]

**Comment by WeiqunZhang** [comment]
> Maybe we can use `The_Comms_Arena()` with gpu-aware MPI, and `The_Arena()` without it.

Yes, that would require the least amount of changes.

File: Src/Particle/AMReX_NeighborParticlesGPUImpl.H [hunk 1]
```diff
@@ -255,6 +255,10 @@ updateNeighborsGPU (bool boundary_neighbors_only)
     }
 
     clearNeighbors();
+
+    snd_buffer.setArena(The_Comms_Arena());
```
Comment (WeiqunZhang) [inline Src/Particle/AMReX_NeighborParticlesGPUImpl.H]: Should these two lines inside `if (ParallelDescriptor::UseGpuAwareMpi())` like in AMReX_ParticleContainerI.H?

Comment (atmyers) [reply Src/Particle/AMReX_NeighborParticlesGPUImpl.H]: Yes.

**Review by WeiqunZhang — APPROVED** [review]

**Comment by ax3l** [comment]
This is causing a memory overhead and performance overhead in ImpactX: https://github.com/ECP-WarpX/impactx/issues/769

Let's discuss next week how we proceed (e.g., opt-in to this PR's features, changing defaults, etc.).

**Pull request merged**